packages/core/src/modules/routing/MediatorModuleConfig.ts

```diff
@@ -9,6 +11,19 @@ export interface MediatorModuleConfigOptions {
    * @default false
    */
   autoAcceptMediationRequests?: boolean
+
+  /**
+   * Strategy to use when a Forward message is received.
+   *
+   *
+   * - `MessageForwardingStrategy.QueueOnly` - simply queue encrypted message into MessagePickupRepository. It will be in charge of manually trigering MessagePickupApi.deliver() afterwards.
+   * - `MessageForwardingStrategy.QueueAndLiveModeDelivery` - Queue message into MessagePickupRepository and deliver it (along any other queued message).
+   * - `MessageForwardingStrategy.DirectDelivery` - Deliver message directly. Do not add into queue (it might be manually added after, e.g. in case of failure)
+   *
+   * @default MessageForwardingStrategy.DirectDelivery
+   * @todo Update default to QueueAndLiveModeDelivery
```

Comment:
When do we want to change the defualt?

Isn't this something that can be determined automatically? E.g. first try direct delivery (in case the mediatee has an endpoint), if it fails, try live delivery (if an open session), if it fails add to queue.


I think you'd often want a combination of 2 and 3 right? Or is direct delivery only direct **live** delivery (I'm assuming it is when you can send it directly to an endpoint, but maybe that's wrong. If so, maybe we should call it `DirectLiveDelivery`)

Reply:
This is more about selecting if we want to encapsulate forwarded messages in Message Pickup's `Delivery` message (so we can have explicit acknowledge) or we want to simply send them to either the mediatee endpoint or an existing session with them. Do you have a better wording to reflect what this setting does?

I think this way of sending forwarded messages should be deprecated at some point, but the encapsulated one should be used only in case the other party supports Message Pickup V2, and that's something that probably can be determined automatically if we do some smarter logic on the agent (e.g. we can query the other party for protocol support or mark it as supported as soon as we receive the first Pickup V2 protocol loop from them).